packages/core/src/cli/perform-build-watch.ts

```diff
@@ -12,14 +12,11 @@ export function performBuildWatch(
 ): void {
   let transformManagerPool = new TransformManagerPool(ts.sys);
   let formatDiagnostic = buildDiagnosticFormatter(ts);
+  let buildProgram = ts.createEmitAndSemanticDiagnosticsBuilderProgram;
 
   let host = ts.createSolutionBuilderWithWatchHost(
     sysForCompilerHost(ts, transformManagerPool),
-    (...args) => {
-      const program = ts.createEmitAndSemanticDiagnosticsBuilderProgram(...args);
-      patchProgram(program, transformManagerPool);
-      return program;
-    },
+    patchProgramBuilder(ts, transformManagerPool, buildProgram),
```

Comment:
Ah, this is good; I kept meaning to do something of this shape, but never figured out the full thing it needed. Glad it's done here!